# Assignment 3: Finish z-test lab and turn it in as a notebook

In [32]:
## follow the UI4_PUI2018 Page 68/76



## Question: is the new Bus route improving commute?

$H_0$:  the commute time is the same or longer with the new bus route as it was before

I will use a sigma level $\alpha$ = 0.05

In [70]:
from __future__ import print_function
import os
import pandas as pd
import numpy as np

puidata = os.getenv("PUIDATA")
if puidata is None:
    os.environ['PUIDATA'] = "%S/PUIDATA"%os.getenv("HOME")
    
print(puidata)



/nfshome/jl9200/PUIdata


In [71]:
## curl ht
# copy file from url to PUIDATA
# 1.first find the link (https://github.com/fedhere/PUI2018_fb55/blob/master/Lab4_fb55/times.txt)
# 2. change the blob to raw and refresh
# 3. link will become (https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt)


!curl https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt --output $PUIDATA/times.txt

    

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0   7357      0 --:--:-- --:--:-- --:--:--  7368


In [72]:

T_N = pd.read_csv(os.getenv("PUIDATA")+'/'+'times.txt',header=None,names=['x'])

T_N.head()

x
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550

In [73]:
T_N_mean = T_N['x'].mean()


In [74]:
T_N.size

100

In [82]:
mu = 36
sigma = 6

zscore = (mu-T_N_mean)/(sigma/np.sqrt(T_N.size))
print ("zscore is {}".format(zscore))

zscore is 2.55639718616667


## Conclusion

#### So, we can see the zscore is 2.56, which is falling outsize of the 2$\sigma$ boundary and we can reject the null hypothesis. 